aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -57,15 +57,25 @@ void div_kernel(TensorIterator& iter) {
       });
     });
   } else {
-    AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "div_cpu", [&]() {
+    if (iter.dtype() == ScalarType::BFloat16) {
       cpu_kernel_vec(iter,
-        [=](scalar_t a, scalar_t b) __ubsan_ignore_float_divide_by_zero__ -> scalar_t {
+        [=](BFloat16 a, BFloat16 b) __ubsan_ignore_float_divide_by_zero__ -> BFloat16 {
            return a / b;
         },
-        [=](Vec256<scalar_t> a, Vec256<scalar_t> b) {
+        [=](Vec256<BFloat16> a, Vec256<BFloat16> b) {
           return a / b;
         });
-    });
+    } else {
+      AT_DISPATCH_FLOATING_TYPES(iter.dtype(), "div_cpu", [&]() {
```

Comment:
Time to introduce `AT_DISPATCH_FLOATING_TYPES_AND` 🤣

Reply:
Yeah, these macros need some rethinking as this approach is not scalable.

Reply:
actually @ifedan added that over in https://github.com/pytorch/pytorch/pull/22433, but this is fine for now.

Reply:
I'm coming here to say, I thought the same thing!

Reply:
Better use that macro (although not perfect either). Unnecessarily duplicated code is worse ;)

Reply:
AT_DISPATCH_FLOATING_TYPES_AND maybe worthwhile for now. The duplication here can be confusing.